pytorch_lightning/metrics/metric.py

```diff
@@ -163,17 +163,27 @@ def aggregate(self, *tensors: torch.Tensor) -> torch.Tensor:
             aggregated values
 
         """
-        try:
-            return torch.cat(tensors).mean(0)
-        except (ValueError, TypeError):
-            if isinstance(tensors[0], Mapping):
-                return {k: torch.stack([tensor[k] for tensor in tensors]).mean(0) for k in tensors[0].keys()}
-            elif isinstance(tensors[0], Sequence) and not isinstance(tensors[0], torch.Tensor):
-                return tuple([torch.stack(tmp).mean(0) for tmp in zip(*tensors)])
-            elif isinstance(tensors[0], torch.Tensor):
-                return torch.stack(tensors).mean(0)
-            else:
-                raise TypeError("unknown metric value format to aggregate")
+        # single tensor
+        if len(tensors) == 1:
+            tensors = tensors[0]
+            if isinstance(tensors, Mapping):
+                return {k: _stack_and_agg(tensors[k]) for k in tensors.keys()}
+            if isinstance(tensors, list):
+                return _stack_and_agg(tensors)
+            if isinstance(tensors, tuple):
+                return tensors
+            if isinstance(tensors, torch.Tensor):
+                return _stack_and_agg(tensors)
```

Comment:
```suggestion
            elif isinstance(tensors, list):
                return _stack_and_agg(tensors)
            elif isinstance(tensors, tuple):
                return tensors
            elif isinstance(tensors, torch.Tensor):
                return _stack_and_agg(tensors)
```

Reply:
@Borda if I change from an `if` to `elif`, codefactor will complain about:
"Unnecessary "elif" after "return"